src/Hug.js

```diff
@@ -1,32 +1,78 @@
 import React from 'react'
+import Box from './Box'
 import Card from './Card'
 import Flex from './Flex'
 import Icon from './Icon'
 import Text from './Text'
 import styled from 'styled-components'
+import { responsiveStyle } from 'styled-system'
 import PropTypes from 'prop-types'
 
-const StyledCard = styled(Card)`
-  border-radius: ${props => props.theme.radius};
+const display = responsiveStyle({
+  prop: 'display',
+  cssProperty: 'display'
+})
+
+const HugCard = styled(Card)`
+  border-top-left-radius: ${props => props.theme.radius};
+  border-top-right-radius: ${props => props.theme.radius};
+  border-bottom-right-radius: 0;
+  border-bottom-left-radius: 0;
+  overflow: hidden;
+`
+
+const BorderConcealer = styled(Box)`
+  z-index: -1;
+  position: relative;
+`
+
+const HideableIcon = styled(Icon)`
+  ${display};
 `
 
-const Hug = props => (
-  <StyledCard {...props} borderColor={props.bg}>
-    <Flex p={3} align="center">
-      {!!props.icon && <Icon mr={3} name={props.icon} size={16} />}
-      <Text.span fontSize={1}>{props.text}</Text.span>
+const Hug = ({
+  showChildBorder,
+  childBorderWidth,
+  bg,
+  color,
+  p,
+  fontSize,
+  icon,
+  iconDisplay,
+  ...props
+}) => (
+  <HugCard {...props} borderColor={bg}>
+    <Flex bg={bg} color={color} p={p} align="center">
+      {!!icon && (
+        <HideableIcon
+          mr={3}
+          mt="-6px"
+          name={icon}
+          size={24}
+          display={iconDisplay}
+        />
+      )}
+      <Text.span fontSize={fontSize}>{props.text}</Text.span>
     </Flex>
-    {props.children}
-  </StyledCard>
+    <BorderConcealer m={showChildBorder ? 0 : `-${childBorderWidth}px`}>
```

Comment:
I think this might be a case where doing something automatic with React.Children might actually be nice since it seems like the most likely use-case. E.g. if `child.type === Card` then remove the border – what do you think?